### ISSUE TITLE
Dataclass field descriptor with default is shown as param's default value in __init__

### ISSUE THREAD

**Issue by debonte** [microsoft/pyright#2990]
**Describe the bug**
If a dataclass field is initialized with dataclasses.field(default=x), when constructing an instance of the class, IntelliSense and hover show the default value as `dataclasses.field(default=x)` rather than `x`. Any other params passed to `field` are also shown, which can result in a lot of visual noise.

**To Reproduce**
Copy the code below and hover over the CustomerModel() call or delete and recreate the parens to see IntelliSense.

**Expected behavior**
id parameter in IntelliSense/hover is shown as `id: int = 5`

**Screenshots or Code**
```python
import dataclasses

@dataclasses.dataclass
class CustomerModel:
    id: int = dataclasses.field(default=5, repr=False, init=True, compare=True, metadata={})

x = CustomerModel()
```
![image](https://user-images.githubusercontent.com/12820898/152623867-ce3ab688-f8b6-415b-b1a1-08b6e1a324f7.png)
![image](https://user-images.githubusercontent.com/12820898/152623876-653608ad-3e50-425d-b729-011f6644cc35.png)

**VS Code extension or command-line**
VS Code with Pylance 2022.2.0

**Comment by erictraut** [comment]
Good catch. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.218, which I just published. It will also be included in the next release of pylance.
